tests/console/zypper_docker.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test zypper-docker installation and its usage
+#    Cover the following aspects of zypper-docker:
+#      * zypper-docker package can be installed
+#      * zypper-docker can list local images:                    'zypper-docker images ls'
+#      * zypper-docker can list updates/patches:                 'zypper-docker list-updates' 'zypper-docker list-patches'
+#      * zypper-docker can list outdated containers:             'zypper-docker ps'
+#      * zypper-docker can list updates/patches for a container: 'zypper-docker list-updates-container' 'zypper-docker list-patches-container'
+#      * zypper-docker can apply the updates:                    'zypper-docker update'
+# Maintainer: Antonio Caristia <acaristia@suse.com>
+
+use base "consoletest";
+use testapi;
+use utils;
+use strict;
+use registration;
+use version_utils qw(is_sle is_leap);
+
+sub run {
+    select_console("root-console");
+
+    install_docker_when_needed();
+
+    # install zypper-docker
+    zypper_call('in zypper-docker');
+
+    # verify zypper-docker is installed
+    validate_script_output("zypper-docker -h", sub { m/zypper-docker - Patching Docker images safely/ }, 180);
+
+    # pull image of openSUSE Leap
+    assert_script_run('docker image pull opensuse:42.3', timeout => 600);
+
+    # list images and check the last one is present
+    my $local_images_list = script_output('docker images');
+    die('docker image opensuse:42.3 not found') unless ($local_images_list =~ /opensuse\s*42.3/);
+
+    # check if 'images ls' works
+    validate_script_output("zypper-docker images ls", sub { m/opensuse/ }, 180);
+
+    # check if 'list-updates' works
+    assert_script_run('zypper-docker list-updates opensuse:42.3', timeout => 600);
+
+    # check if 'list-patches' works
+    assert_script_run('zypper-docker list-patches opensuse:42.3', timeout => 600);
+
+    # run a container
+    assert_script_run("docker container run -d --name tmp_container opensuse:42.3 tail -f /dev/null");
+
+    # check if 'ps' works
+    assert_script_run("zypper-docker ps", timeout => 600);
+
+    # check if 'list-updates-container' works
+    assert_script_run("zypper-docker list-updates-container tmp_container", timeout => 600);
+
+    # check if 'list-patches-container' works
+    assert_script_run("zypper-docker list-patches-container tmp_container", timeout => 600);
+
+    # apply all the updates to a new_image
+    assert_script_run("zypper-docker update --auto-agree-with-licenses opensuse:42.3 new_image", timeout => 600);
+}
```

Comment:
I think this test module is pretty clean and easy to read. However it also looks a bit verbose. Probably many of the comments are redundant, the blank lines in between can be deleted and you could extract `opensuse:42.3` which we would need to replace in some time anyway with a variable.